Now, we have all of the LINCS small molecules (SMs) labelled with ICD-11 labels. We can take just the first character to classify each of the SMs but first we need to find out how many of those we have with ICD-11 labels, both `unique` and `redundant`.

In [1]:
import pandas as pd

# Define the path of the 'mapped_TTD_LINCS_df.csv' file
mapped_file_path = '/data4/msc23104470/TTD/mapped_TTD_LINCS_df.csv'
mapped_file = pd.read_csv(mapped_file_path, sep=',')

In [2]:
mapped_file.columns

Index(['rid', 'PSME1', 'ATF1', 'RHEB', 'FOXO3', 'RHOA', 'IL1B', 'ASAH1',
       'RALA', 'ARHGEF12',
       ...
       'CID', 'PUBCHSID', 'CASNUMBE', 'CHEBI_ID', 'SUPDRATC', 'SUPDRCAS',
       'DRUGNAME_y', 'INDICATI', 'ICD-11', 'CLINICAL_STATUS'],
      dtype='object', length=994)

In [3]:
# Drop the unnecessary columns from the dataframe
mapped_file.drop(columns=['PUBCHSID', 'CASNUMBE', 'CHEBI_ID', 'SUPDRATC', 'SUPDRCAS', 'DRUGNAME_y', 'INDICATI', 'CLINICAL_STATUS'], inplace=True)

In [5]:
mapped_file.columns

Index(['rid', 'PSME1', 'ATF1', 'RHEB', 'FOXO3', 'RHOA', 'IL1B', 'ASAH1',
       'RALA', 'ARHGEF12',
       ...
       'MLLT11', 'CEBPZ', 'CBR3', 'ATC Codes', 'InChI Key', 'TTDDRUID',
       'DRUGNAME_x', 'D_FOMULA', 'CID', 'ICD-11'],
      dtype='object', length=986)

In [7]:
# Drop 'ATC Codes', 'InChI Key', 'TTDRUID', 'DRUGNAME_x', 'D_FOMULA', 'CID'..
mapped_file.drop(columns=['ATC Codes', 'InChI Key', 'TTDDRUID', 'DRUGNAME_x', 'D_FOMULA', 'CID'], inplace=True)

In [8]:
mapped_file.columns

Index(['rid', 'PSME1', 'ATF1', 'RHEB', 'FOXO3', 'RHOA', 'IL1B', 'ASAH1',
       'RALA', 'ARHGEF12',
       ...
       'STK10', 'UGDH', 'SQRDL', 'HEBP1', 'ATP11B', 'CD320', 'MLLT11', 'CEBPZ',
       'CBR3', 'ICD-11'],
      dtype='object', length=980)

In [22]:
mapped_file

rid     PSME1      ATF1  \
0                  AML001_CD34_6H:BRD-K43389675:10  0.004217 -0.018644   
1                  AML001_CD34_6H:BRD-K43389675:10  0.004217 -0.018644   
2                  AML001_CD34_6H:BRD-K43389675:10  0.004217 -0.018644   
3                   AML001_PC3_6H:BRD-A45664787:10 -0.000762 -0.000233   
4              AML001_PC3_6H:BRD-K27316855:0.37037 -0.002200  0.006644   
...                                            ...       ...       ...   
14280  RAD001_PC3_6H:BRD-K84937637-001-03-2:0.1235 -0.001343  0.022330   
14281  RAD001_PC3_6H:BRD-K84937637-001-03-2:0.1235 -0.001343  0.022330   
14282      RAD001_PC3_6H:BRD-K84937637-001-03-2:10  0.006401  0.009406   
14283      RAD001_PC3_6H:BRD-K84937637-001-03-2:10  0.006401  0.009406   
14284      RAD001_PC3_6H:BRD-K84937637-001-03-2:10  0.006401  0.009406   

           RHEB     FOXO3      RHOA      IL1B     ASAH1      RALA  ARHGEF12  \
0      0.019958 -0.056607 -0.017211 -0.040410 -0.012175 -0.013127 -0.006405   
1      0.019958 -0.056607 -0.017211 -0.040410 -0.012175 -0.013127 -0.006405   
2      0.019958 -0.056607 -0.017211 -0.040410 -0.012175 -0.013127 -0.006405   
3      0.003371  0.004516  0.009972  0.130947 -0.024684  0.017885  0.023541   
4     -0.010455  0.034244 -0.000161 -0.015078 -0.024642 -0.033437 -0.010307   
...         ...       ...       ...       ...       ...       ...       ...   
14280  0.011539  0.005044  0.017594  0.009275 -0.007178 -0.028721  0.033979   
14281  0.011539  0.005044  0.017594  0.009275 -0.007178 -0.028721  0.033979   
14282  0.021147 -0.014808  0.013453 -0.008992  0.003342 -0.006246 -0.005264   
14283  0.021147 -0.014808  0.013453 -0.008992  0.003342 -0.006246 -0.005264   
14284  0.021147 -0.014808  0.013453 -0.008992  0.003342 -0.006246 -0.005264   

       ...     STK10      UGDH     SQRDL     HEBP1    ATP11B     CD320  \
0      ...  0.012359  0.028994  0.035868  0.012153  0.036431  0.001324   
1      ...  0.012359  0.028994  0.035868  0.012153  0.036431  0.001324   
2      ...  0.012359  0.028994  0.035868  0.012153  0.036431  0.001324   
3      ...  0.029266  0.038909 -0.007869  0.029858  0.012888 -0.052345   
4      ... -0.037752  0.012920  0.031202  0.002686  0.006412 -0.021728   
...    ...       ...       ...       ...       ...       ...       ...   
14280  ...  0.000216  0.012335  0.009009  0.012971  0.001469  0.013327   
14281  ...  0.000216  0.012335  0.009009  0.012971  0.001469  0.013327   
14282  ... -0.012605 -0.043669  0.015122  0.041121  0.009549  0.048752   
14283  ... -0.012605 -0.043669  0.015122  0.041121  0.009549  0.048752   
14284  ... -0.012605 -0.043669  0.015122  0.041121  0.009549  0.048752   

         MLLT11     CEBPZ      CBR3     ICD-11  
0      0.039140 -0.055924 -0.012614       2A60  
1      0.039140 -0.055924 -0.012614       2B57  
2      0.039140 -0.055924 -0.012614  2A00-2F9Z  
3     -0.008905  0.045641 -0.023661     BB01.0  
4      0.020117  0.019046  0.026748        NaN  
...         ...       ...       ...        ...  
14280 -0.008807 -0.037837  0.022568       NE84  
14281 -0.008807 -0.037837  0.022568       8D64  
14282 -0.012588 -0.029582  0.010061       2A83  
14283 -0.012588 -0.029582  0.010061       NE84  
14284 -0.012588 -0.029582  0.010061       8D64  

[14285 rows x 980 columns]

We have a total of 980 columns in our dataframe now.

column 1, column 2, column 3, ... column n-1, column n

`SM, Gene 1, Gene 2, Gene 3, ... Gene N, ICD-11`(label)

In [11]:
mapped_file.isnull().sum()

rid          0
PSME1        0
ATF1         0
RHEB         0
FOXO3        0
          ... 
CD320        0
MLLT11       0
CEBPZ        0
CBR3         0
ICD-11    8585
Length: 980, dtype: int64

There are `8585` null entries in the last column

In [12]:
mapped_file.notnull().sum()

rid       14285
PSME1     14285
ATF1      14285
RHEB      14285
FOXO3     14285
          ...  
CD320     14285
MLLT11    14285
CEBPZ     14285
CBR3      14285
ICD-11     5700
Length: 980, dtype: int64

That means there are 5700 labelled small molecules.

Now, we have to check how many of small molecules are unique in our dataset. To do this, we first need to 

i) drop those molecules that have no ICD-11 labels and then,

ii) extract the BRD Codes from the first column and add that as the last column of our dataframe. Then, we can use `unique()` to find out how many of them are infact unique!

#### using dropna method in pandas

In [13]:
last_column_name = mapped_file.columns[-1]
df_cleaned = mapped_file.dropna(subset=[last_column_name])

In [16]:
# Recheck for null entries in the cleaned dataframe
df_cleaned.isnull().sum()

rid       0
PSME1     0
ATF1      0
RHEB      0
FOXO3     0
         ..
CD320     0
MLLT11    0
CEBPZ     0
CBR3      0
ICD-11    0
Length: 980, dtype: int64

In [17]:
import re

# Define a function to extract the BRD- ID
def extract_brd_id(rid):
    match = re.search(r'BRD-[A-Za-z0-9]{9}', rid)
    if match:
        return match.group(0)
    return None

# Apply the function to the 'rid' column and create a new column 'BRD_id'
df_cleaned['BRD'] = df_cleaned['rid'].apply(extract_brd_id)

/tmp/ipykernel_33992/2203631365.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_cleaned['BRD'] = df_cleaned['rid'].apply(extract_brd_id)


In [21]:
# now check how many of small molecules are infact unique..
unique_count = df_cleaned['BRD'].nunique()
print(f"We have {unique_count} unique small molecules in the dataframe.")

We have 468 unique small molecules in the dataframe.


In [23]:
df_cleaned

rid     PSME1      ATF1  \
0                  AML001_CD34_6H:BRD-K43389675:10  0.004217 -0.018644   
1                  AML001_CD34_6H:BRD-K43389675:10  0.004217 -0.018644   
2                  AML001_CD34_6H:BRD-K43389675:10  0.004217 -0.018644   
3                   AML001_PC3_6H:BRD-A45664787:10 -0.000762 -0.000233   
7              AML001_PC3_6H:BRD-K43389675:0.37037  0.004243 -0.019425   
...                                            ...       ...       ...   
14280  RAD001_PC3_6H:BRD-K84937637-001-03-2:0.1235 -0.001343  0.022330   
14281  RAD001_PC3_6H:BRD-K84937637-001-03-2:0.1235 -0.001343  0.022330   
14282      RAD001_PC3_6H:BRD-K84937637-001-03-2:10  0.006401  0.009406   
14283      RAD001_PC3_6H:BRD-K84937637-001-03-2:10  0.006401  0.009406   
14284      RAD001_PC3_6H:BRD-K84937637-001-03-2:10  0.006401  0.009406   

           RHEB     FOXO3      RHOA      IL1B     ASAH1      RALA  ARHGEF12  \
0      0.019958 -0.056607 -0.017211 -0.040410 -0.012175 -0.013127 -0.006405   
1      0.019958 -0.056607 -0.017211 -0.040410 -0.012175 -0.013127 -0.006405   
2      0.019958 -0.056607 -0.017211 -0.040410 -0.012175 -0.013127 -0.006405   
3      0.003371  0.004516  0.009972  0.130947 -0.024684  0.017885  0.023541   
7     -0.009953  0.009629  0.003882  0.026752  0.007623 -0.001681  0.003181   
...         ...       ...       ...       ...       ...       ...       ...   
14280  0.011539  0.005044  0.017594  0.009275 -0.007178 -0.028721  0.033979   
14281  0.011539  0.005044  0.017594  0.009275 -0.007178 -0.028721  0.033979   
14282  0.021147 -0.014808  0.013453 -0.008992  0.003342 -0.006246 -0.005264   
14283  0.021147 -0.014808  0.013453 -0.008992  0.003342 -0.006246 -0.005264   
14284  0.021147 -0.014808  0.013453 -0.008992  0.003342 -0.006246 -0.005264   

       ...      UGDH     SQRDL     HEBP1    ATP11B     CD320    MLLT11  \
0      ...  0.028994  0.035868  0.012153  0.036431  0.001324  0.039140   
1      ...  0.028994  0.035868  0.012153  0.036431  0.001324  0.039140   
2      ...  0.028994  0.035868  0.012153  0.036431  0.001324  0.039140   
3      ...  0.038909 -0.007869  0.029858  0.012888 -0.052345 -0.008905   
7      ...  0.023102 -0.041017  0.012306  0.010878 -0.004279  0.025922   
...    ...       ...       ...       ...       ...       ...       ...   
14280  ...  0.012335  0.009009  0.012971  0.001469  0.013327 -0.008807   
14281  ...  0.012335  0.009009  0.012971  0.001469  0.013327 -0.008807   
14282  ... -0.043669  0.015122  0.041121  0.009549  0.048752 -0.012588   
14283  ... -0.043669  0.015122  0.041121  0.009549  0.048752 -0.012588   
14284  ... -0.043669  0.015122  0.041121  0.009549  0.048752 -0.012588   

          CEBPZ      CBR3     ICD-11            BRD  
0     -0.055924 -0.012614       2A60  BRD-K43389675  
1     -0.055924 -0.012614       2B57  BRD-K43389675  
2     -0.055924 -0.012614  2A00-2F9Z  BRD-K43389675  
3      0.045641 -0.023661     BB01.0  BRD-A45664787  
7      0.042857  0.044424       2A60  BRD-K43389675  
...         ...       ...        ...            ...  
14280 -0.037837  0.022568       NE84  BRD-K84937637  
14281 -0.037837  0.022568       8D64  BRD-K84937637  
14282 -0.029582  0.010061       2A83  BRD-K84937637  
14283 -0.029582  0.010061       NE84  BRD-K84937637  
14284 -0.029582  0.010061       8D64  BRD-K84937637  

[5700 rows x 981 columns]

In [25]:
df_cleaned.drop(columns=['BRD'], inplace=True)

/tmp/ipykernel_33992/968857268.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_cleaned.drop(columns=['BRD'], inplace=True)


In [27]:
# Now drop the BRD (last column).
df_cleaned.to_csv('/data4/msc23104470/data-matrix-TTD-w-HUGO-symbols-ICD-11.csv', sep=',')